cherrypy/lib/sessions.py

```diff
@@ -578,15 +578,16 @@ def clean_up(self):
         """Clean up expired sessions."""
         now = self.now()
         # Iterate over all session files in self.storage_path
-        for fname in os.listdir(self.storage_path):
+        for entry in os.scandir(self.storage_path):
```

Comment:
If we're changing this, let's go for the higher level `pathlib` interface already rather then working with the low-level `os` API.

Reply:
Surprisingly, when I attempted later on to switch from using os.scandir() to pathlib's iterdir(), the performance significantly deteriorated. I am currently using a Windows server, and the calls to os.listdir() or path.iterdir() hardly ever return, whereas scandir() was able to return and start iterating within approximately 30 seconds. So I would suggest sticking to os.scandir() for better performance.

Reply:
`pathlib` is the interface of choice for maintainability reasons. Also, using file-based sessions where never intended for high-performance use-cases. Having a 30-second scan suggests that other file system operations during normal runtime aren't exactly fast either.
So I'd say it's an architecture/deployment problem on the app side, not the framework. A typical solution is to use session storages that fit better for this purpose — like memcached or redis.

Could you post the timings/metrics showing the difference? And perhaps, report this issue on the upstream CPython tracker?

Reply:
I'm inclined to say we should accept this change as proposed as it provides a demonstrated performance benefit. The problem, as webknjaz has illustrated is that a subsequent developer might propose to change it from scandir to Path.iterdir, and there's nothing in the test suite that would prevent that. That is, if CherryPy wishes to support this use-case, and my instinct is that it should if possible (and jiangwen's experience shows it is possible), there should be a test to capture the expectation that this change sets. That is, let's not support the use-case incidentally, but intentionally, and add a test for it.